controllers/chat/getSignedChannelList.js

```diff
@@ -73,10 +73,18 @@ const __filterAndTransformChannelData = async (channel) => {
   if (messageLength === 0 && channelType === CHANNEL_TYPE.TOPIC) return false;
 
   if (channelType === CHANNEL_TYPE.TOPIC) {
-    const topic = await TopicFunction.findOneByName(Topics, removePrefixTopic(newChannel.id));
-    newChannel.image = topic.icon_path || null;
-    newChannel.channelImage = topic.icon_path || null;
-    newChannel.channel_image = topic.icon_path || null;
+    const topic = await TopicFunction.getTopicDetailByTopicId(
+      sequelize,
+      userId,
+      removePrefixTopic(newChannel.id)
+    );
+
+    newChannel.image = topic?.icon_path || null;
+    newChannel.channelImage = topic?.icon_path || null;
+    newChannel.channel_image = topic?.icon_path || null;
+    newChannel.cover_image = topic?.cover_path || null;
+    newChannel.is_following = topic?.is_following || false;
+    newChannel.topic_follower_count = parseInt(topic?.topic_follower_count || 0);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The way topic details are fetched has been changed. Instead of using `findOneByName`, it now uses `getTopicDetailByTopicId`. Also, more properties have been added to `newChannel`. Ensure that these changes do not introduce any inconsistencies in how topics are handled elsewhere in your application.
```diff
-    const topic = await TopicFunction.findOneByName(Topics, removePrefixTopic(newChannel.id));
-    newChannel.image = topic.icon_path || null;
-    newChannel.channelImage = topic.icon_path || null;
-    newChannel.channel_image = topic.icon_path || null;
+    const topic = await TopicFunction.getTopicDetailByTopicId(
+      sequelize,
+      userId,
+      removePrefixTopic(newChannel.id)
+    );
+    newChannel.image = topic?.icon_path || null;
+    newChannel.channelImage = topic?.icon_path || null;
+    newChannel.channel_image = topic?.icon_path || null;
+    newChannel.cover_image = topic?.cover_path || null;
+    newChannel.is_following = topic?.is_following || false;
+    newChannel.topic_follower_count = parseInt(topic?.topic_follower_count || 0);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->